pkg/backend/telemetryapi/harvester.go

```diff
@@ -257,14 +290,15 @@ func postData(req *http.Request, client *http.Client) response {
 	if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusAccepted {
 		r.body, _ = ioutil.ReadAll(resp.Body)
 	} else {
+		_, _ = ioutil.ReadAll(resp.Body)
```

Comment:
🤔

Reply:
It is good practice to read the entire `Body` and close it so we can reuse the underlying TCP connection. Some more details [here](https://golang.org/pkg/net/http/#Client.Do).

Reply:
and `resp.Body.Close()` wouldn't work?

Reply:
You need to also read all the content as well so that the TCP connection can be reused. We do a `resp.Body.Close()` in a defer on L281 but only ever read the body if we get a 200 or 201. This means the body for any other status code will be closed but we never read it to EOF before closing it.

Reply:
Gotcha! TY